plugins/c9.core/util.js

```diff
@@ -402,10 +402,15 @@ define(function(require, exports, module) {
                 setTimeout(callback, 17);
             };
         
-        plugin.freezePublicAPI({});
+        plugin.freezePublicAPI({
+            get supportedIcons(){ return SupportedIcons; },
+            set supportedIcons(value){ SupportedIcons = value; },
+            get contentTypes(){ return contentTypes; },
+            set contentTypes(value){ contentTypes = value; }
```

Comment:
I think we shouldn't turn this into public api this way, because 1. we do not really use contentTypes for anything, and it makes api unnecessarily complicated, 2. this looks like something that will cause conflicts between plugins trying to modify it. For which file types do you want to add icons? adding them to default theme too might be an easier workaround.